cmd/kubeadm/app/cmd/upgrade/node.go

```diff
@@ -206,7 +206,7 @@ func (d *nodeData) IgnorePreflightErrors() sets.Set[string] {
 	return d.ignorePreflightErrors
 }
 
-// KubeconfigPath returns the path to the user kubeconfig file.
+// KubeConfigPath returns the path to the user kubeconfig file.
```

Comment:
strange, are we not running a linter on these packages - i.e. to be able to catch the godoc problems?

Reply:
It seems that the staticcheck's behavior is inconsistent with golint. 

https://github.com/golang/lint

Reply:
```sh
$ cd cmd/kubeadm/app/constants
$ golint
constants.go:425:2: comment on exported const KubeControllerManagerUserName should be of the form "KubeControllerManagerUserName ..."
constants.go:427:2: comment on exported const KubeAPIServerUserName should be of the form "KubeAPIServerUserName ..."
constants.go:429:2: comment on exported const KubeSchedulerUserName should be of the form "KubeSchedulerUserName ..."
constants.go:431:2: comment on exported const EtcdUserName should be of the form "EtcdUserName ..."
constants.go:433:2: comment on exported const ServiceAccountKeyReadersGroupName should be of the form "ServiceAccountKeyReadersGroupName ..."

$ staticcheck
<no output>
```

Reply:
hm, i was under the impression we are running a suite of linters on k/k, which covers the godoc exports:
https://github.com/kubernetes/kubernetes/blob/master/hack/verify-golangci-lint.sh

cc @pohly @dims

Reply:
We run golangci-lint with a pretty restricted set of enabled checks: https://github.com/kubernetes/kubernetes/blob/master/hack/golangci.yaml

See https://groups.google.com/a/kubernetes.io/g/dev/c/myGiml72IbM/m/aQTb6KToAwAJ for an effort to enable additional linters.

However, enforcing this particular godoc convention is not planned. It's too contentious and often not necessary.

Reply:
thanks for the info, i guess we are going to have to run the godoc checks out of band / accept PRs that try to fix these.

Reply:
So you want "godocs" enabled for `cmd/kubeadm`?

Once https://github.com/golangci/golangci-lint/pull/3617 is merged, it should be possible to enable additional, stricter checks for specific directories that are not enabled elsewhere. Right now that is possible, but very awkward.

Reply:
yes, we have to skip the kubeadm api packages, but it would be nice to enable it for the rest.